include/flox/resolver/descriptor.hh

```diff
@@ -155,6 +169,10 @@ FLOX_DEFINE_EXCEPTION( ParseManifestDescriptorRawException,
 
 /**
  * @brief A set of user defined requirements describing a package/dependency.
+ *
+ * May either be defined as a set of attributes or with a string matching
+ * this syntax:
+ * `[<input>:]((<attr>.)*<attrName>|<pname>)[@(<semver>|=<version>)]`
```

Comment:
question nonblocking: I don't understand the `(<attr>.)*<attrName>|<pname>)` part of this. Not sure how important it is to get it right, but my understanding is closer to `((<attr>.)+<attrName>)|(<pname>|<attrName>|<name>)`